### PR TITLE
Improve XXE attack module

### PR DESCRIPTION
**Pull request by devl00p**

### CODE REVIEW

**Comment by devl00p** [comment]
Two improvements : 
* Add two more XXE payloads (one for Unix, one for Windows) that doesn't use php filters
* Generate a medium vulnerability report when the DTD was reached but no data was exfiltrated

wapiti_xxe

**Comment by devl00p** [comment]
Fix #81

**Comment by devl00p** [comment]
Exfiltration tested against https://pentesterlab.com/exercises/play_xxe/course (Java based app)

File: endpoint/get_xxe.php [hunk 1]
```diff
@@ -14,6 +14,7 @@
     foreach (glob("$directory/*/*/*.txt") as $filename) {
         $parts = explode("/", $filename);
         list($time, $payload_num, $ip) = explode("-", basename($filename, ".txt"));
+	    // ./xxe_data/<session_id>/<path_id>/<hex_param>/<filename>
```
Comment (JulienTD) [inline endpoint/get_xxe.php:17]: maybe remove this comment or rearrange it to follow the other lines  indentation and a description

File: endpoint/xxe_dtd.php [hunk 2]
```diff
@@ -14,26 +14,48 @@
...
 }
 
+$time = time();
+$directory = "./xxe_data/$session_id/$path_id/$hex_param/";
+if (!is_dir($directory)) mkdir($directory, 0777, true);
+
+// Create an empty file so we can know if the DTD was queried
+touch("$directory/$time-$payload_num-$ip.txt");
+
+if (substr($payload, 0, 4 ) === "java") {
```
Comment (JulienTD) [inline endpoint/xxe_dtd.php:49]: there is an extra space after the 4, should be: `if (substr($payload, 0, 4) === "java") {`

**Review by JulienTD — CHANGES_REQUESTED** [review]

**Pull request merged**